internal/util/util.go

```diff
@@ -338,14 +339,9 @@ func GetModTime(path string) (time.Time, error) {
 	return info.ModTime(), nil
 }
 
-// EscapePath replaces every path separator in a given path with a %
+// EscapePath encodes the path in URL query form
 func EscapePath(path string) string {
-	path = filepath.ToSlash(path)
-	if runtime.GOOS == "windows" {
-		// ':' is not valid in a path name on Windows but is ok on Unix
-		path = strings.ReplaceAll(path, ":", "%")
-	}
-	return strings.ReplaceAll(path, "/", "%")
+	return url.QueryEscape(filepath.ToSlash(path))
```

Comment:
I already saw this complaint coming...

1. Correct me in case I'm wrong, but your proposal isn't backward compatible too. Existing backups wouldn't be loaded since the expected path/name doesn't fit to the current, since `%` aren't stored yet with `%%`.
2. My proposal uses a scheme which is common enough to be known by technically skilled people and can be reverse interpret on file level to reconstruct the real paths/names. The original scheme can only be adapted in case `micro`'s code is checked and even then there are ambiguities.

How this should be solved in a general and bulletproof way without breaking with the compatibility?
Sometimes we've to jump over our own shadow.

Reply:
>    1. Correct me in case I'm wrong, but your proposal isn't backward compatible too. Existing backups wouldn't be loaded since the expected path/name doesn't fit to the current, since `%` aren't stored yet with `%%`.

In the corner case when the file name contains `%` (or some directory name in its path contains `%`), yes, the existing backup wouldn't be loaded. But in this case it is not loaded anyway, which is precisely the issue we are trying to fix. So backward compatibility would not be broken, right? (Although it's true that the issue would not be fixed for existing backups, it would only be fixed for new backups.)

Yeah, there is no way to fix this issue in a fully backward compatible way, precisely because the existing format is ambiguous. (And if it wasn't ambiguous, there would be no need to fix it.)

But with your solution, each and every existing backup would not be loaded, not just those with `%` in the file path.

>    2. My proposal uses a scheme which is common enough to be known by technically skilled people and can be reverse interpret on file level to reconstruct the real paths/names. The original scheme can only be adapted in case `micro`'s code is checked and even then there are ambiguities.

I'm not sure what are you trying to say here. (OTOH, I'm not sure if that is important.)

I would be happy to use URL encoding, but we should care about users, so unfortunately we need to care about compatibility.

Reply:
One more downside of changing the format in an incompatible way is that old backups will not be removed, they will stay in the user's `~/.config/micro/backups` forever as a garbage. (Unless the user notices and removes them manually; assuming that the user is aware (LOL) that this is an old unsupported format, so these are backups for some old irrelevant files and can be removed.)

Reply:
Also note that this ambiguous `%` encoding is used not only for backups but also for files in `~/.config/micro/buffers`.

There is a funny thing I recently realized: those files are never removed. Right now I have 4364 files in my `~/.config/micro/buffers`, starting from files created back in 2020. (I have `savecursor` turned on.) The total size of my `~/.config/micro/buffers` is 19MB.

Reply:
BTW a good test case for files with `%` in their names: open a backup file in micro. I.e. testing a backup for a backup (for a backup for a backup...).

Reply:
> I'm not sure what are you trying to say here. (OTOH, I'm not sure if that is important.)

What I was trying to say was the following:
In case we apply the proposed encoding we are able to reconstruct the full path from file name perspective to e.g. restore the backup to his origin path via core utils instead of `micro` itself.
At least this is something I'd like to be able...

> I would be happy to use URL encoding, but we should care about users, so unfortunately we need to care about compatibility.

I'll have to accept it with gritted teeth and once again a new `TODO` hopefully not forget in the moment we're ready to do breaking changes (e.g. 3.x).

> One more downside of changing the format in an incompatible way is that old backups will not be removed, they will stay in the user's `~/.config/micro/backups` forever as a garbage.

The ones with ambiguities anyway and we only get rid of all of this stuff in the moment we do these necessary breaking changes AND inform the users about these.
AFAIK the release notes have been invented for such a purpose. :wink: 

So according to semantic version scheme not before the major 3.x.

Reply:
> In case we apply the proposed encoding we are able to reconstruct the full path from file name perspective to e.g. restore the backup to his origin path via core utils instead of `micro` itself.
At least this is something I'd like to be able...

```
echo "$filename" | sed -e 's:%:/:g' -e 's://:%:g'
```

Actually... this has just made me realize that encoding `%` with `%%`, which I suggested, would be still ambiguous:

```
$ echo '%foo%%%bar' | sed -e 's:%:/:g' -e 's://:%:g'
/foo%/bar
```

but it could as well be `/foo/%bar`.

So we need something better.

Reply:
Hmm... actually we could support both URL encoding and the legacy % encoding at the same time? We convert the file name (which we know) to the backup name, not the other way around. So we can try converting it using both encodings, and if either file exists in `~/.config/micro/backups`, apply it as a backup. (And if both exist, the URL encoded one should probably have higher priority.)

Although there is a risk of a conflict causing applying a backup to a wrong file, if the user opens `/foo/bar` and there is a backup file `%2Ffoo%2Fbar` but it is actually a legacy-named backup file for `/2Ffoo/2Fbar`. But the likelihood of such conflicts in real life is probably negligibly low.

Reply:
Yes, we only convert in one direction (non-encoded -> encoded).
Sounds like a feasible plan! :+1: